### PR TITLE
fuzzgen: Disable unimplemented riscv64 ops

### PR DESCRIPTION
**Pull request by afonso360**
👋 Hey,

This is a WIP PR to disable all the missing and broken instructions in fuzzgen for the riscv64 backend.

### CODE REVIEW

**Review by elliottt — APPROVED** [review]
This looks great! Looking forward to have riscv64 enabled for fuzzing :D

File: cranelift/fuzzgen/src/function_generator.rs [hunk 1]
```diff
@@ -457,6 +457,125 @@ fn valid_for_target(triple: &Triple, op: Opcode, args: &[Type], rets: &[Type]) -
...
+                (Opcode::Rotl, &[I128, I16]),
+                (Opcode::Rotl, &[I128, I32]),
+                (Opcode::Rotl, &[I128, I64]),
+                (Opcode::Rotl, &[I128, I128]),
+                // https://github.com/bytecodealliance/wasmtime/issues/5525
+                (Opcode::Sextend, &[I8], &[I128]),
+                (Opcode::Sextend, &[I16], &[I128]),
+                (Opcode::Sextend, &[I32], &[I128]),
+                (Opcode::Sextend, &[I64], &[I128]),
+                (Opcode::Sextend, &[I128], &[I128]),
```
Comment (jameysharp) [inline cranelift/fuzzgen/src/function_generator.rs:494]: As an observation for any of us working on this list in the future, exceptions like these can be abbreviated by using more wildcards:

```suggestion
                // https://github.com/bytecodealliance/wasmtime/issues/5523
                (Opcode::Rotl, &[I128, _]),
                // https://github.com/bytecodealliance/wasmtime/issues/5525
                (Opcode::Sextend, _, &[I128]),
```

I don't think that's important, I just wanted to mention it while merging this.

**Review by jameysharp — APPROVED** [review]
Very exciting, thank you!

File: cranelift/fuzzgen/src/function_generator.rs (hunk 1 above)
Comment (afonso360) [reply cranelift/fuzzgen/src/function_generator.rs:494]: Oh, I didn't know that! That's really nice!

**Comment by jameysharp** [comment]
@uweigand, FYI: This PR illustrates what we need to start routinely fuzz-testing additional Cranelift backends. I'd love to start testing s390x on our OSS-Fuzz quota too. For non-x86, OSS-Fuzz can only test that codegen succeeds, not that the generated code runs correctly, but I expect that to be enough to catch many bugs. Of particular interest right now, this should help us catch non-SSA lowerings.

The steps are:
1. add the `s390x` feature to the `cranelift-codegen` dependency in `fuzz/Cargo.toml` (optionally, you can temporarily remove the other backends there to focus on that one)
2. run `cargo +nightly fuzz run cranelift-icache` on any host, and/or `cargo +nightly fuzz run cranelift-fuzzgen` on an s390x host—whichever you have more CPU cycles available on
3. as it finds bugs, add backend-specific exceptions for known issues with lowering particular instructions at particular types, as illustrated by this PR

Once the fuzzer runs for an hour without finding new bugs it's probably ready.

These fuzz targets don't yet generate all possible Cranelift instructions, so you won't have to write down exceptions for every unimplemented thing. It's currently limited to scalar types and arithmetic/load/store operations mostly, but we're incrementally expanding coverage.

**Comment by uweigand** [comment]
@jameysharp s390x support PR now posted here: https://github.com/bytecodealliance/wasmtime/pull/5596

Thanks for pointing this out!

**Pull request merged**